app/src/main/java/com/hover/stax/transfers/TransferFragment.kt

```diff
@@ -277,7 +279,7 @@ class TransferFragment : AbstractFormFragment(), ActionSelect.HighlightListener,
         val accountError = accountsViewModel.errorCheck()
         payWithDropdown.setState(accountError, if (accountError == null) AbstractStatefulInput.SUCCESS else AbstractStatefulInput.ERROR)
 
-        val amountError = transferViewModel.amountErrors()
+        val amountError = transferViewModel.amountErrorsCheck(actionSelectViewModel.getAmountValidationRegex())
```

Comment:
Improve consistency by making this api look like the recipient error check. Just pass the selected action itself to the error check method. This also separates concerns better - only the error check method needs to know how to get the validation regex.